canvas_manage_course/settings/base.py

```diff
@@ -60,14 +60,18 @@
               "into SchoolPermissions the lti_permission entry can be removed "
               "from INSTALLED_APPS.", DeprecationWarning)
 
+# Note: ordering of cache-related middleware is important, see
+# https://docs.djangoproject.com/en/1.8/topics/cache/#order-of-middleware-classes
 MIDDLEWARE_CLASSES = (
+    'django.middleware.cache.UpdateCacheMiddleware',
     'djangular.middleware.DjangularUrlMiddleware',
     'django.contrib.sessions.middleware.SessionMiddleware',
     'django.middleware.common.CommonMiddleware',
     'cached_auth.Middleware',
     'django_auth_lti.middleware_patched.MultiLTILaunchAuthMiddleware',
     'django.contrib.messages.middleware.MessageMiddleware',
     'django.middleware.security.SecurityMiddleware',
+    'django.middleware.cache.FetchFromCacheMiddleware',
```

Comment:
I'm not crazy about introducing two new middlewares here just to add some headers to the response, esp. since the main purpose of these middlewares is to provide server-side caching of generated content.  Adding the `@never_cache` decorator as needed seems like a simpler, more targeted approach to me.